core/options_list.go

```diff
@@ -0,0 +1,151 @@
+package core
+
+import (
+	"strings"
+
+	containerlablabels "github.com/srl-labs/containerlab/labels"
+	"github.com/srl-labs/containerlab/types"
+)
+
+// ListOption is a type used for functional options for the Clab List method.
+type ListOption func(o *ListOptions)
+
+// ListOptions represents the options for listing containers.
+type ListOptions struct {
+	labName                 string
+	nodeName                string
+	toolType                string
+	containerlabLabelExists bool
+	cliArgs                 []string
+}
+
+// NewListOptions returns a new destroy options object.
+func NewListOptions() *ListOptions {
+	return &ListOptions{}
+}
+
+func (o *ListOptions) ToFilters() []*types.GenericFilter {
```

Comment:
and here is the hopefully nicer way to simplify/centralize the filter creation stuff